### ISSUE TITLE
feature-request:  Log additional context when a policy fails to monitor

### ISSUE THREAD

**Issue by baxor** [hashicorp/nomad-autoscaler#507]
If a policy is misconfigured, it may issue an error in the log similar to this: 
`2021-07-01T17:13:02.470Z [ERROR] policy_manager.policy_handler: policy validation failed: policy_id=0b28c559-686f-dd0c-0464-4a60e1175e09 errors=["scaling.policy.check must be []interface{}, found map[string]interface {}", "scaling.policy.check must be []interface{}, found map[string]interface {}"]`

But there is no context as to which job or task the policy came from, which makes it difficult to track down in large deployments.  The error is emitted from [here](https://github.com/hashicorp/nomad-autoscaler/blob/main/policy/nomad/source.go#L242), however it was not immediately obvious in this function or its calling function how to access a desirable context (job/task name) to submit a PR.

**Comment by jrasell** [comment]
Hi @baxor, this is certainly something we should look at improving. Thanks for taking the time to raise this issue.

**Comment by josegonzalez** [comment]
Would a PR that includes more information about the scaling policy be accepted? Much as I'd love to not write code, I also love not having to spend an hour digging through a mess of log output to figure out which scaling event is associated with which policy we wrote.

**Comment by jrasell** [comment]
@josegonzalez yes we would certainly love and appreciate a PR on this. Please reach out if you have any questions.
